### ISSUE TITLE
Convert to TypeScript

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/beers-law-lab#287]
Requested by @kathy-phet.

This also includes Concentration.

**Comment by pixelzoom** [comment]
Baselines published before starting this work:

https://phet-dev.colorado.edu/html/beers-law-lab/1.7.0-dev.5
https://phet-dev.colorado.edu/html/concentration/1.6.0-dev.1

**Comment by pixelzoom** [comment]
Conversion of Concentraion to TypeScript as completed in the 2 commits above.

**Comment by pixelzoom** [comment]
Conversion to TypeScript is done.

@kathy-phet FYI a large chunk of the conversion (4.5 hours) was specific to PhET-iO.  This is the first sim I've converted to TypeScript that makes heavy use of IOTypes and PhetioGroup.  And I discovered that those parts of PhET-iO do not have the same level of TypeScript maturity as other parts of the PhET code base.  After working with @zepumph and @samreid for close to an hour, I was able to get past the major landmines, and call the sim "converted".   But those parts of PhET-iO and this sim are not in great shape, and will need to be revisited. 

Closing.
